alpha/veneer/basic/basic.go

```diff
@@ -0,0 +1,83 @@
+package basic
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/pkg/image"
+)
+
+type BasicVeneer struct {
+	Registry image.Registry
+}
+
+func (v BasicVeneer) Ingest(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	// only taking first argument as file
+	stat, serr := os.Stat(ref)
+	if serr != nil {
+		return nil, serr
+	}
+
+	if stat.IsDir() {
+		return nil, errors.New("cannot ingest veneers by directory reference") // BUGBUG enhance to walk and eat all files in turn, heeding ignore file
+	}
+	return v.renderFile(ctx, ref)
+}
+
+func (v BasicVeneer) renderFile(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	var cfg = &declcfg.DeclarativeConfig{}
+
+	// xform any relative to absolute paths
+	abspath, err := filepath.Abs(ref)
+	if err != nil {
+		return nil, err
+	}
+	// xform to break apart dir/file elements
+	rpath, fname := filepath.Split(abspath)
+
+	root := os.DirFS(rpath)
+	file, err := root.Open(fname)
+	if err != nil {
+		return nil, err
+	}
+
+	defer file.Close()
+
+	if cfg, err = declcfg.LoadFile(root, file, ref); err != nil {
+		return cfg, err
+	}
+
+	var outb = cfg.Bundles[:0] // allocate based on max size of input, but empty slice
+	var r action.Render
+	// populate registry, incl any flags from CLI
+	if v.Registry != nil {
+		r.Registry = v.Registry
+	}
```

Comment:
This can be simplified to:
```suggestion
	// populate registry, incl any flags from CLI
	r := action.Render{
		Registry: v.Registry
	}
```

Reply:
I'd also suggest enforcing that we only allow rendering bundle images, by setting `AllowedRefMask: action.RefBundleImage` on the `action.Render` struct